src/uct/cuda/cuda_copy/cuda_copy_iface.c

```diff
@@ -260,30 +322,32 @@ static UCS_CLASS_INIT_FUNC(uct_cuda_copy_iface_t, uct_md_h md, uct_worker_h work
         return UCS_ERR_IO_ERROR;
     }
 
-    self->stream_d2h = 0;
-    self->stream_h2d = 0;
+    uct_iface_set_async_event_params(params, &self->async.event_cb,
+                                     &self->async.event_arg);
 
-    ucs_queue_head_init(&self->outstanding_d2h_cuda_event_q);
-    ucs_queue_head_init(&self->outstanding_h2d_cuda_event_q);
+    for (i = 0; i < UCT_CUDA_COPY_STREAM_LAST; i++) {
+        self->stream[i]          = 0;
+        ucs_queue_head_init(&self->outstanding_event_q[i]);
+    }
 
     return UCS_OK;
 }
 
 static UCS_CLASS_CLEANUP_FUNC(uct_cuda_copy_iface_t)
 {
     int active;
+    int i;
 
     UCT_CUDADRV_CTX_ACTIVE(active);
 
     uct_base_iface_progress_disable(&self->super.super,
                                     UCT_PROGRESS_SEND | UCT_PROGRESS_RECV);
     if (active) {
-        if (self->stream_h2d != 0) {
-            UCT_CUDA_FUNC_LOG_ERR(cudaStreamDestroy(self->stream_h2d));
-        }
-
-        if (self->stream_d2h != 0) {
-            UCT_CUDA_FUNC_LOG_ERR(cudaStreamDestroy(self->stream_d2h));
+        for (i = 0; i < UCT_CUDA_COPY_STREAM_LAST; i++) {
+            if (self->stream[i] != 0) {
```

Comment:
why would it be 0?

Reply:
https://github.com/openucx/ucx/blob/41d3ad06f7775c2a861ceccc73e1fd801d1d9098/src/uct/cuda/cuda_copy/cuda_copy_iface.c#L329

We manually set that to zero during iface init

Reply:
can it really stay 0 in the dtor?

Reply:
These streams get created on the first attempt to move data using cudaMemcpy. If cuda_copy iface initialization takes place but no endpoint operations occur, then streams never get created and remain set to zero.

Reply:
do we really need to check both `if (active) {` and `if (self->stream[i] != 0) {` ?